content/intro-to-storybook/svelte/en/test.md

```diff
@@ -0,0 +1,154 @@
+---
+title: 'Test UI components'
+tocTitle: 'Testing'
+description: 'Learn the ways to test UI components'
+---
+
+No Storybook tutorial would be complete without testing. Testing is essential to creating high quality UIs. In modular systems, miniscule tweaks can result in major regressions. So far we encountered three types of tests:
+
+- **Visual tests** rely on developers to manually look at a component to verify it for correctness. They help us sanity check a component’s appearance as we build.
+- **Snapshot tests** with Storyshots capture a component’s rendered markup. They help us stay abreast of markup changes that cause rendering errors and warnings.
+- **Unit tests** with Jest verify that the output of a component remains the same given an fixed input. They’re great for testing the functional qualities of a component.
+
+## “But does it look right?”
+
+Unfortunately, the aforementioned testing methods alone aren’t enough to prevent UI bugs. UIs are tricky to test because design is subjective and nuanced. Visual tests are too manual, snapshot tests trigger too many false positives when used for UI, and pixel-level unit tests are poor value. A complete Storybook testing strategy also includes visual regression tests.
+
+## Visual regression testing for Storybook
+
+Visual regression tests are designed to catch changes in appearance. They work by capturing screenshots of every story and comparing them commit-to-commit to surface changes. This is perfect for verifying graphical elements like layout, color, size, and contrast.
+
+<video autoPlay muted playsInline loop style="width:480px; margin: 0 auto;">
+  <source
+    src="/intro-to-storybook/visual-regression-testing.mp4"
+    type="video/mp4"
+  />
+</video>
+
+Storybook is a fantastic tool for visual regression testing because every story is essentially a test specification. Each time we write or update a story we get a spec for free!
+
+There are a number of tools for visual regression testing. For professional teams we recommend [**Chromatic**](https://www.chromaticqa.com/), an addon made by Storybook maintainers that runs tests in the cloud.
+
+## Setup visual regression testing
+
+Chromatic is a hassle-free Storybook addon for visual regression testing and review in the cloud. Since it’s a paid service (with a free trial), it may not be for everyone. However, Chromatic is an instructive example of a production visual testing workflow that we'll try out for free. Let’s have a look.
+
+### Initialize local repository
+
+At this stage there's still no repository setup just yet. Let's change that and set it up:
+
+```bash
+$ git init 
+```
+
+Next add files to the first commit.
+
+```bash
+$ git add -A
+```
+
+Now commit the files.
+
+```bash
+$ git commit -m "taskbox UI"
+```
+
+### Get Chromatic
+
+Add the package as a dependency.
+
+```bash
+npm install -D storybook-chromatic
+```
+
+Import Chromatic in your `.storybook/config.js` file.
```

Comment:
This step isn't needed any more. We should probably remove it from all tutorials (cc @ndelangen)

Reply:
Indeed, this should be removed from all tutorials.

Reply:
It's now removed with the latest commit. Going to mark it as resolved if you both don't mind